skyportal/handlers/api/source.py

```diff
@@ -643,10 +650,30 @@ def get(self, obj_id):
               application/json:
                 schema: Error
         """
-        source = Source.get_obj_if_owned_by(obj_id, self.current_user)
+        source = Source.get_obj_if_owned_by(
+            obj_id,
+            self.current_user,
+            options=[joinedload(Source.obj).joinedload(Obj.photometry)],
+        )
         if source is None:
             return self.error('Invalid source ID.')
 
+        initial_pos = (
+            source.ra_dis if source.ra_dis is not None else source.ra,
+            source.dec_dis if source.dec_dis is not None else source.dec,
```

Comment:
The discovery RA and Dec will often correspond to some of the most astrometrically uncertain photometry. Why not default to the highest S/N detection as the fall back coordinate?

Reply:
I'd prefer to keep this at the position when the source is first saved (ie. passes a filter).